services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/permissions/Authorizer.kt

```diff
@@ -0,0 +1,37 @@
+package fi.hel.haitaton.hanke.permissions
+
+import fi.hel.haitaton.hanke.HankeNotFoundException
+import fi.hel.haitaton.hanke.HankeRepository
+import fi.hel.haitaton.hanke.currentUserId
+import org.springframework.stereotype.Component
+import org.springframework.transaction.annotation.Transactional
+
+abstract class Authorizer(
+    private val permissionService: PermissionService,
+    private val hankeRepository: HankeRepository? = null
+) {
+    fun authorize(hankeId: Int?, permissionCode: PermissionCode, ex: () -> Exception) {
+        hankeId
+            ?.let { permissionService.hasPermission(it, currentUserId(), permissionCode) }
+            .let { if (it != true) throw ex() }
+    }
+
+    internal fun authorizeHankeTunnus(
+        hankeTunnus: String,
+        permissionCode: PermissionCode
+    ): Boolean {
+        val hankeIds = hankeRepository!!.findOneByHankeTunnus(hankeTunnus)
+        authorize(hankeIds?.id, permissionCode) { HankeNotFoundException(hankeTunnus) }
+        return true
+    }
+
+    @Transactional(readOnly = false)
+    open fun authorizeHankeTunnus(hankeTunnus: String, permissionCode: String): Boolean =
```

Comment:
`permissionCode` is a String because the Spring Expression Language `@PreAuthorize` uses can only handle enums if given the full path, like:
```
    @PreAuthorize( 
        "@hankeAuthorizer.authorizeHankeTunnus(#hankeTunnus,T(fi.hel.haitaton.hanke.permissions.PermissionCode).EDIT)"
    )
```
I think this is quite unwieldy, so I thought I would use the names instead. These are all static values, so any mistakes should come up in tests.